shm/inc/map.hpp

```diff
@@ -2,10 +2,14 @@
 
 #include <vector>
 
+#include "coordinates.hpp"
 #include "island.hpp"
 
 class Map {
 private:
     std::vector<Island> islands_;
     Island* currentPosition_;
+
+public:
+    Island* getIsland(const Coordinates&);
```

Comment:
also, Mateusz said that's good to add name of the parameter in function declaration:
```suggestion
    Island* getIsland(const Coordinates& coordinates);
```